lib/stat_tracker.rb

```diff
@@ -303,4 +362,26 @@ def fewest_goals_scored(team_id)
   def average_win_percentage(team_id)
     win_percentage_per_team[team_id]
   end
+
+  def win_percentage_by_season(team_id)
+    result = {}
+    @game_teams.each do |game_team|
+      result[game_team.season] = win_percentage_per_team
+    end
+    result
+  end
+
+  def worst_season(team_id)
+    lowest_win_percent = win_percentage_by_season(team_id).values.min
+    win_percentage_by_season(team_id).invert[lowest_win_percent]
+  end
+
+  def favorite_opponent(team_id)
+    find_team(opponents_and_num_result_for_team(team_id, "WIN").max_by {|opponent, num_result| num_result}[0]).name
+  end
+
+  def rival(team_id)
+    find_team(opponents_and_num_result_for_team(team_id, "LOSS").max_by {|opponent, num_result| num_result}[0]).name
+  end
+
```

Comment:
I submitted a bug issue for this, but we'll have to update favorite_opponent and rival to calculate based on max/min win percentage against a given team vs. max/min of pure num of wins. This is my bad! We can handle through this PR or separately via that bug issue.